### ISSUE TITLE
Problems with Mail System and environment specific settings

### ISSUE THREAD

**Issue by AdamGerthel** [oddhill/odddrupal#21]
The [Mail System](https://www.drupal.org/project/mailsystem) module is seemingly very old (it even lacks version nr!). I tried configuring different mail system settings per environment in order to send production e-mails via the [SMTP module](https://www.drupal.org/project/smtp) and quickly realized that the settings variables used by Mail System were behaving in a very odd manner. Settings seem to be saved as system variables, but not loaded as such, making it impossible to control via Features or Tadaa. I then noticed that there are quite a lot of differences between the current version in Odd Drupal and the latest official release (which is from 2012).

I tested the latest official 2.x version and I wasn't very impressed with that one either - it saves the configuration in files in files/mailsystem which won't work for different environments either. Maybe this is just a module that would need a custom module that handles the settings, I'm not sure. Anyway, a module overlook might be needed.

**Comment by AdamGerthel** [comment]
Also, the patch from #2 in issue [#1844630 ](https://www.drupal.org/node/1844630) should probably be added.

**Comment by olofbokedal** [comment]
Some initial information. The current version being used is the one from commit http://cgit.drupalcode.org/mailsystem/commit/?id=4c957ae. This was done because the previous update (7.x-3.0-alpha1) introduced some bugs which didn't exist in 4c957ae. So the change was basically a downgrade in order to avoid some bugs.

I'll investigate which version that we should update to (2.x or 3.x). The patch mentioned has been commited to 7.x-3.x-dev so I'll aim for that one.

**Comment by olofbokedal** [comment]
Not sure if the commit on Odd Drupal actually represented the 4c957ae commit though, since the only change between the current version and the latest 7.x dev is the LICENSE.txt file...

The patch mentioned earlier has probably already been implemented, but I'll investigate the variables issues a bit. Perhaps this will lead to issues at the project, but we'll see.

**Comment by olofbokedal** [comment]
@AdamGerthel are you trying to change the settings for SITE-WIDE DEFAULT MAIL SYSTEM? It seems as if they are stored among the other settings, in a variable called mail_system, which is an array with different settings.

Isn't it possible to control this variable with Tadaa or Features?

**Comment by AdamGerthel** [comment]
@olofjohansson yes and now. I am trying to set site wide default mail system. And no, it isn't really set in the variable you are referring to - or rather it is, but sometimes only for a short amount of time. Try it ourself and try changing it via Tadaa to another value, and visit the settings page again. I'm not sure why it acts so strange, I haven't looked at the module code.
